libsql-replication/proto/metadata.proto

```diff
@@ -16,4 +16,6 @@ message DatabaseConfig {
     optional string jwt_key = 7;
     optional uint64 txn_timeout_s = 8;
     bool allow_attach = 9;
+    optional bool shared_schema = 10;
```

Comment:
Why not just bool with default false instead of optional?

Reply:
How would it behave if we parse existing database config, where this tag was not defined? Wouldn't it error?

Reply:
if there's a default value then missing field should get default value.

Reply:
but this is not crucial and should block this PR.